test/metrics/metrics-factory.ts

```diff
@@ -81,5 +94,97 @@ contract(
 			})
 		})
+		describe('MetircsFactory; destroy', () => {
+			const dev = new DevProtocolInstance(deployer)
+			let metricsAddress1: string
+			let metricsAddress2: string
+			before(async () => {
+				await dev.generateAddressConfig()
+				await Promise.all([
+					dev.generateMarketGroup(),
+					dev.generateMetricsFactory(),
+					dev.generateMetricsGroup()
+				])
+				await dev.addressConfig.setMarketFactory(marketFactory)
+				await dev.marketGroup.addGroup(market, {from: marketFactory})
+				const metricsFactoryResult1 = await dev.metricsFactory.create(
+					property1,
+					{from: market}
+				)
+				const metricsFactoryResult2 = await dev.metricsFactory.create(
+					property1,
+					{from: market}
+				)
+				metricsAddress1 = getMetricsAddress(metricsFactoryResult1)
+				metricsAddress2 = getMetricsAddress(metricsFactoryResult2)
+			})
+			it('If a non-metrics address is specified, an error will occur.', async () => {
+				const result = await dev.metricsFactory
+					.destroy(dummyMetrics, {
+						from: market
+					})
+					.catch((err: Error) => err)
+				validateErrorMessage(result, 'address is not metrics')
+			})
+			it('Cannot be executed from other than market contract.', async () => {
+				const result = await dev.metricsFactory
+					.destroy(metricsAddress1, {
+						from: user
+					})
+					.catch((err: Error) => err)
+				validateAddressErrorMessage(result)
+			})
+			it('If a metrics address is specified, the remove process is performed and the event is also issued.', async () => {
+				let result = await dev.metricsGroup.isGroup(metricsAddress1, {
+					from: deployer
+				})
+				expect(result).to.be.equal(true)
+				await dev.metricsFactory.destroy(metricsAddress1, {
+					from: market
+				})
+				result = await dev.metricsGroup.isGroup(metricsAddress1, {
+					from: deployer
+				})
+				expect(result).to.be.equal(false)
+				const [from, metrics] = await new Promise<string[]>(resolve => {
+					watch(dev.metricsFactory, WEB3_URI)('Destroy', (_, values) => {
+						const {_from, _metrics} = values
+						resolve([_from, _metrics])
+					})
+				})
+				expect(market).to.be.equal(from)
+				expect(metricsAddress1).to.be.equal(metrics)
+			})
+			it('You can also run the destroy method in owner.', async () => {
+				let result = await dev.metricsGroup.isGroup(metricsAddress2, {
+					from: deployer
+				})
+				expect(result).to.be.equal(true)
+				await dev.metricsFactory.destroy(metricsAddress2, {
+					from: deployer
+				})
+				result = await dev.metricsGroup.isGroup(metricsAddress2, {
+					from: deployer
+				})
+				expect(result).to.be.equal(false)
+				const [from, metrics] = await new Promise<string[]>(resolve => {
+					watch(dev.metricsFactory, WEB3_URI)('Destroy', (_, values) => {
+						const {_from, _metrics} = values
+						resolve([_from, _metrics])
+					})
+				})
+				expect(deployer).to.be.equal(from)
+				expect(metricsAddress2).to.be.equal(metrics)
+			})
```

Comment:
I think this spec is too dangerous.

Reply:
Is it because it can be done from outside the market contract?

Reply:
This spec is expected to be removed as soon as possible as a temporary thing.